packages/workflow/src/Workflow.ts

```diff
@@ -1421,6 +1424,74 @@ export class Workflow {
 			};
 		}
 	}
+
+	/**
+	 * Select the node to use as starter for a subworkflow execution.
+	 */
+	static selectSubworkflowStarter(nodes: INode[]) {
+		const found = nodes.find((n) => SUBWORKFLOW_STARTER_NODES.includes(n.type));
+
+		if (found) return found;
+
+		const starterNodesToDisplay = SUBWORKFLOW_STARTER_NODES.join(', ');
+
+		throw new ApplicationError(
```

Comment:
It used to throw `SubworkflowOperationError` or `CliWorkflowOperationError` - now we're always throwing `ApplicationError`. Won't this break stuff?

Reply:
- `CliWorkflowOperationError` is [never checked for](https://github.com/search?q=repo%3An8n-io%2Fn8n%20CliWorkflowOperationError&type=code) in `master`.
- Restored `SubWorkflowOperationError` but this added a circular dependency... Let me see how to untangle but it might not be trivial.